### ISSUE TITLE
The HAL ObjectMapper no more produces valid HAL representation if the JaxbAnnotationModule is installed

### ISSUE THREAD

**Issue by jjoslet** [spring-projects/spring-hateoas#783]
**Affect versions:** 0.23.0, 0.25.0

When adding the `com.fasterxml.jackson.module.jaxb.JaxbAnnotationModule` Jackson Module in the HAL `ObjectMapper`, the Spring HATEOAS HAL representation is incorrect.

In the attached [sample project](https://github.com/spring-projects/spring-hateoas/files/2741549/sample-no-autoconfigure.zip), instead of having the HAL response:
```
{
    "content": "Hello, World!",
    "_links": {
        "self": {
            "href": "http://localhost:8080/greeting?name=World"
        }
    }
}
```
I get:
```
{
    "content": "Hello, World!",
    "link": {
        "self": {
            "href": "http://localhost:8080/greeting?name=World"
        }
    }
}
```
`link` instead of `_links`.

See the [sample project](https://github.com/spring-projects/spring-hateoas/files/2741549/sample-no-autoconfigure.zip) to reproduce. This sample is based on https://github.com/spring-guides/gs-rest-hateoas/tree/master/complete.

Here is a linked discussion in the Spring Boot project : https://github.com/spring-projects/spring-boot/issues/15663

**Comment by gregturn** [comment]
I see it as well. Even when I upgrade to Spring HATEOAS 0.25.0.RELEASE.

Suspect that the ordering of Jackson modules is making things go wonky. Trying to verify so I know how to fix it.

**Comment by gregturn** [comment]
It's taken a little digging, but it appears this module may be acting as designed.

`JaxbAnnotationModule` is designed to let you use JAXB annotations to drive Jackson. In essence, reuse the POJOs you defined for your old SOAP services while you build new JSON-based REST services.

Spring HATEAOAS 0.25.x still has JAXB annotations...

```java
abstract class ResourceSupportMixin extends ResourceSupport {

	@Override
	@XmlElement(name = "link")
	@JsonProperty("_links")
	@JsonInclude(Include.NON_EMPTY)
	@JsonSerialize(using = Jackson2HalModule.HalLinkListSerializer.class)
	@JsonDeserialize(using = Jackson2HalModule.HalLinkListDeserializer.class)
	public abstract List<Link> getLinks();
}
```

So it's taking it's cue from the `@XmlElement` annotation over the `@JsonProperty` one. In many cases this would be okay. But for HAL it's dead wrong.

HAL JSON uses `_links` while HAL XML uses `<link ...>`.

Thus, adding `JaxbAnnotationModule` is a request to use Spring HATEOAS, but leveraging its JAXB support, which is actually not good. We've never supported HAL XML and never supported this combination. This is born out by the removal of JAXB annotations in 1.0 via #695.

If you're looking to support two different services in your application, i.e. one that DOES use this approach while ALSO using Spring HATEOAS in a completely different service, you may need to register this module directly on the ObjectMapper relevant to your other service.

In short, this is a configuration we cannot support.
